### ISSUE TITLE
initContainer metrics not working on v1.12.7 or v1.13.5

### ISSUE THREAD

**Issue by jmcmeek** [kubernetes/kubernetes#76292]
**What happened**:
`kubectl` top pods does not show pods having initContainers.  The `alb` pods below have an init container and are not shown `top pods`:
```
$ kubectl -n kube-system get pod | grep alb
public-cr002e2963213f421fbc0cac5e9175f92c-alb1-845d6b54-2ks9c   4/4     Running   0          2d22h
public-cr002e2963213f421fbc0cac5e9175f92c-alb1-845d6b54-c5pkd   4/4     Running   0          2d22h

$ kubectl -n kube-system top pod | grep alb
$
```

The metrics-server has log entries like
```
E0408 19:55:34.671477       1 manager.go:102] unable to fully collect metrics: [unable to fully scrape metrics from source kubelet_summary:10.177.223.164: unable to get CPU for container "sysctl" in pod kube-system/public-cr002e2963213f421fbc0cac5e9175f92c-alb1-845d6b54-c5pkd on node "10.177.223.164", discarding data: missing cpu usage metric, unable to fully scrape metrics from source kubelet_summary:10.177.223.139: unable to get CPU for container "sysctl" in pod kube-system/public-cr002e2963213f421fbc0cac5e9175f92c-alb1-845d6b54-2ks9c on node "10.177.223.139", discarding data: missing cpu usage metric]
```
I see similar results for other pods having init containers.  Other pods are shown.

I see that kubelet `/stats/summary` is missing `usageNanoCores` for the init container.
```
      "containers": [
        {
          "name": "ingress-auth-1",
          "startTime": "2019-04-05T21:45:06Z",
          "cpu": {
            "time": "2019-04-08T20:21:20Z",
            "usageNanoCores": 4766,
            "usageCoreNanoSeconds": 7167923443
          },
output skipped
        {
          "name": "sysctl",
          "startTime": "2019-04-05T21:44:41Z",
          "cpu": {
            "time": "2019-04-08T20:21:26Z",
            "usageCoreNanoSeconds": 0
          },
```

**What you expected to happen**:

`metrics-server` should return metrics for pods that have init containers.

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7", GitCommit:"6f482974b76db3f1e0f5d24605a9d1d38fad9a2b", GitTreeState:"clean", BuildDate:"2019-03-25T02:52:13Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12", GitVersion:"v1.12.7+IKS", GitCommit:"2de3526b7279db4c3141299ec252c55c94be84ac", GitTreeState:"clean", BuildDate:"2019-04-01T08:08:00Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}

$ kubectl get node
NAME             STATUS   ROLES    AGE   VERSION
10.177.223.139   Ready    <none>   67d   v1.12.7+IKS
10.177.223.164   Ready    <none>   67d   v1.12.7+IKS
10.177.223.178   Ready    <none>   67d   v1.12.7+IKS
```

I replicated these results with Kubernetes 1.13.5.

- Cloud provider or hardware configuration:
  IBM Cloud Kubernetes Service
  Container runtime: containerd 1.2.4 and 1.1.6
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

**Comment by jmcmeek** [comment]
We first saw this on Kubernetes 1.12.6 and I found https://github.com/kubernetes/kubernetes/pull/74336 that seemed to address this problem and was cherry picked to 1.11 - 1.13.  But we see the same problem in 1.12.7 which has the cherry-pick.

The cherry-picks are different from the original PR in that `result.CPU.UsageNanoCores` is not set in the cherry-picks.  Does that need to be set, presumably in a different way than in the original PR?

**Comment by jmcmeek** [comment]
I tried adding the `result.CPU.UsageNanoCores` assignments to 1.13.5 and building hyperkube.  That appears to fix the problem.   Seems odd that those lines were dropped from the cherry-picks.

**Comment by jmcmeek** [comment]
Shall I turn my work into a PR?  If so, start with release-1.13?

**Comment by jmcmeek** [comment]
I think we can close this issue.  All the PRs that I expect to see merged have been merged already.  I'm checking to see what to do with the remaining 1.11 PR.

**Comment by mrksngl** [comment]
I don't think this issue is actually resolved, I got the same problem on 1.21.1.

I used this as mwe:
```yaml
apiVersion: v1
kind: Pod
metadata:
  name: test
spec:
  initContainers:
  - name: init
    image: busybox:latest
    imagePullPolicy: IfNotPresent
    command:
    - sleep
    - "1"
  containers:
  - name: pause
    image: k8s.gcr.io/pause
```

after waiting a bit for metrics to be collected, running `kubectl top pod test` says
> Error from server (NotFound): podmetrics.metrics.k8s.io "default/test" not found

If I leave the `initContainer` away, metrics are available shortly after.

I'm using cri-o 1.21.0 as container environment.

**Comment by roy-work** [comment]
I also see this issue, on 1.19. Can the bug be re-opened?
